.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateTimeAltExtractor.cs

```diff
@@ -367,42 +352,118 @@ private bool IsConnectorOrWhiteSpace(int start, int end, string text)
-                        if (match.Success)
+                        if (contextEr == null)
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            extractResults[i].Data = GenerateAltData(extractResults[i].Type, parentText, contextEr: null);
+                            extractResults[i].Type = ExtractorName;
+
+                            isValidContext = true;
+                        }
+
+                        int j = i + 1;
+
+                        while (j < extractResults.Count)
                         {
-                            var contextErs = new ExtractResult
+                            var relativePrefixContext = ExtractRelativePrefixContext(extractResults[j].Text);
+
+                            // No relativePrefix extracted, the context would follow the previous relativePrefix
+                            // Such as "Wednesday" in "next Tuesday or Wednesday"
+                            if (relativePrefixContext == null)
+                            {
+                                extractResults[j].Data = GenerateAltData(extractResults[j].Type, parentText, contextEr);
+                                extractResults[j].Type = ExtractorName;
+                                j++;
+                            }
+                            else
                             {
-                                Text = match.Value,
-                                Start = match.Index,
-                                Length = match.Length,
-                                Type = Constants.ContextType_RelativePrefix
-                            };
-
-                            data.Add(Constants.Context, contextErs);
-                            data.Add(Constants.SubType, Constants.SYS_DATETIME_DATE);
+                                // Current extraction has relativePrefix, the context would not follow the previous relativePrefix
+                                // Such as "Wednesday" in "next Monday or Tuesday or previous Wednesday"
+                                contextEr = relativePrefixContext;
+                                break;
+                            }
                         }
+
+                        i = j;
                     }
                 }
             }
-            
+
+            return isValidContext;
+        }
+
+        private Dictionary<string, object> GenerateAltData(string subType, string parentText, ExtractResult contextEr = null)
+        {
+            var data = new Dictionary<string, object>();
+
+            data.Add(Constants.SubType, subType);
+            data.Add(ExtendedModelResult.ParentTextKey, parentText);
+
+            if (contextEr != null)
+            {
+                data.Add(Constants.Context, contextEr);
+            }
+
             return data;
         }
 
+
+        private ExtractResult ExtractRelativePrefixContext(string text)
```

Comment:
Which text is this? The whole query? Can you rename for clarity?

Reply:
Change this parameter name to "entityText"